src/components/Dataset/GeneratedUI.js

```diff
@@ -62,84 +62,90 @@ const GeneratedUI = ({settings, ui, changeSettingsCallback}) => {
-                        }
-                        break;
-                    case 'select': {
-                        let options = [];
-                        // Build options
-                        data.options.forEach(v => {
-                            // <Select> does not like to display text if the key or value (?) is null.
-                            // So convert null to string "- not selected -".
-                            if (v === null || v === '') {
-                                // console.log("pushing: ", {
-                                //     key: "- not selected -",
-                                //     value: "- not selected -",
-                                //     text: "- not selected -"
-                                // });
-                                options.push({
-                                    key: "- not selected -",
-                                    value: "- not selected -",
-                                    text: "- not selected -"
-                                })
-                            } else {
-                                // console.log("pushing: ", {key: v, value: v, text: v});
-                                options.push({key: v, value: v, text: v})
+                                    console.log(settings[ui_name])
+                                    let newSettings = {...settings};
+                                    // Convert from "not selected" string back to null.
+                                    if (data.value === "- not selected -") {
+                                        newSettings[ui_name] = null;
+                                    } else {
+                                        newSettings[ui_name] = data.value;
+                                    }
+                                    // Special dropdown logic for selection mechanism
+                                    if (ui_name === "selection_mechanism") {
+                                        // Hide other selection mechanisms and show current one
+                                        for (const option of data.options) {
+                                            newSettings[option.value] = null;
+                                            ui['data'][option.value]['shown'] = false;
+                                        }
+                                        ui['data'][data.value]['shown'] = true;
+                                        // Update new settings with default value of selected mechanism
+                                        newSettings[data.value] = ui['data'][data.value]['options'][0]
+                                    }
+                                    changeSettingsCallback(newSettings)
+                                }}
+                                selection
+                            />);
+                            // TODO: add a onChange that dispatches the setting (do this for int and string as well)
```

Comment:
Does this need to be addressed for this PR?

Reply:
probably not needed for now